From 33f60fe2c64272cf1472aa541bcb491a62f81ca5 Mon Sep 17 00:00:00 2001
From: Naveen Saini <naveen.kumar.saini@intel.com>
Date: Mon, 26 Oct 2020 16:24:09 +0800
Subject: [PATCH] Use standard installation directory variables

Upstream-Status: Submitted [https://github.com/intel/vc-intrinsics/pull/35]

Signed-off-by: Naveen Saini <naveen.kumar.saini@intel.com>
---
 GenXIntrinsics/CMakeLists.txt | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/GenXIntrinsics/CMakeLists.txt b/GenXIntrinsics/CMakeLists.txt
index c59c589..5bcf0d1 100644
--- a/GenXIntrinsics/CMakeLists.txt
+++ b/GenXIntrinsics/CMakeLists.txt
@@ -33,7 +33,7 @@ endif()
 # this option is to switch on install when we are building not inside IGC
 if(INSTALL_REQUIRED)
   install(DIRECTORY include/llvm
-    DESTINATION include
+    DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
     COMPONENT genx-intrinsics-headers
     FILES_MATCHING
     PATTERN "*.h"
@@ -43,7 +43,7 @@ if(INSTALL_REQUIRED)
   # and then "install(DIRECTORY" installs them even if they are empty
   # so generated file has to be installed separetely
   install(FILES ${CMAKE_CURRENT_BINARY_DIR}/include/llvm/GenXIntrinsics/${GENX_INTRINSICS_DESCRIPTION}
-    DESTINATION include/llvm/GenXIntrinsics
+    DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/llvm/GenXIntrinsics
     COMPONENT genx-intrinsics-headers
   )
 
@@ -52,15 +52,15 @@ if(INSTALL_REQUIRED)
   # we need to limit it to external build
   if(BUILD_EXTERNAL)
     install(TARGETS LLVMGenXIntrinsics EXPORT LLVMGenXIntrinsicsTargets
-      LIBRARY DESTINATION lib
-      ARCHIVE DESTINATION lib
-      RUNTIME DESTINATION bin
-      INCLUDES DESTINATION include
+      LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+      ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+      RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+      INCLUDES DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
     )
 
     install(EXPORT LLVMGenXIntrinsicsTargets
       FILE LLVMGenXIntrinsicsConfig.cmake
-      DESTINATION lib/cmake/LLVMGenXIntrinsics
+      DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/LLVMGenXIntrinsics
     )
   endif() # BUILD_EXTERNAL
 endif() # INSTALL_REQUIRED
-- 
2.17.1

