From a36a4788a52128e5ccbec4cd29eb167153dcc8b8 Mon Sep 17 00:00:00 2001
From: Diana Chen <po-yu.chen@intel.com>
Date: Mon, 11 Jan 2021 19:06:36 +0000
Subject: [PATCH] IGA: Add missing header

Upstream-Status: Backport
Signed-off-by: Anuj Mittal <anuj.mittal@intel.com>
---
 visa/iga/IGALibrary/IR/SWSBSetter.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/visa/iga/IGALibrary/IR/SWSBSetter.cpp b/visa/iga/IGALibrary/IR/SWSBSetter.cpp
index 87f119f3b..6956edf0d 100644
--- a/visa/iga/IGALibrary/IR/SWSBSetter.cpp
+++ b/visa/iga/IGALibrary/IR/SWSBSetter.cpp
@@ -27,6 +27,7 @@ SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 #include "Traversals.hpp"
 #include "BitSet.hpp"
 #include <iterator>
+#include <limits>
 using namespace iga;
 
 /**
-- 
2.37.3

