From 5329f00de484b470152e2bd3a5ab6dac45668b13 Mon Sep 17 00:00:00 2001
From: Anuj Mittal <anuj.mittal@intel.com>
Date: Thu, 19 May 2022 22:50:09 +0800
Subject: [PATCH] fix tblgen

---
 IGC/cmake/igc_llvm.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/IGC/cmake/igc_llvm.cmake b/IGC/cmake/igc_llvm.cmake
index 89c03fc49..85be9ae08 100644
--- a/IGC/cmake/igc_llvm.cmake
+++ b/IGC/cmake/igc_llvm.cmake
@@ -48,7 +48,7 @@ else()
   set(LLVM_OPT_EXE "opt" CACHE STRING "")
 
   set(LLVM_TABLEGEN_EXE "llvm-tblgen")
-  if(CMAKE_CROSSCOMPILING)
+  if(TRUE)
     if(DEFINED LLVM_TABLEGEN)
       set(LLVM_TABLEGEN_EXE ${LLVM_TABLEGEN})
     else()
-- 
2.35.3

