From ac19bd37abccb697cc5fc9647be48de14185e9a2 Mon Sep 17 00:00:00 2001
From: Dongwon Kim <dongwon.kim@intel.com>
Date: Tue, 1 Mar 2022 15:23:36 -0800
Subject: [PATCH] find external llvm-tblgen new

Signed-off-by: Dongwon Kim <dongwon.kim@intel.com>
---
 IGC/cmake/igc_llvm.cmake | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/IGC/cmake/igc_llvm.cmake b/IGC/cmake/igc_llvm.cmake
index 038a9ce3e..80c83d4e0 100644
--- a/IGC/cmake/igc_llvm.cmake
+++ b/IGC/cmake/igc_llvm.cmake
@@ -24,7 +24,10 @@ set(CMAKE_MODULE_PATH
   ${CMAKE_MODULE_PATH}
   )
 
-set(LLVM_TABLEGEN_EXE "llvm-tblgen")
+find_program(LLVM_TABLEGEN_EXE "llvm-tblgen")
+if(LLVM_TABLEGEN_EXE-NOTFOUND)
+  message(FATAL_ERROR "[VC] llvm-tblgen is not found")
+endif()
 
 set(LLVM_LINK_EXE "llvm-link" CACHE STRING "")
 
-- 
2.20.1

