include cstdint header for gcc13

Upstream-Status: Pending
Signed-off-by: Anuj Mittal <anuj.mittal@intel.com>

diff --git a/src/gpu/jit/gemm/kernel_catalog.hpp b/src/gpu/jit/gemm/kernel_catalog.hpp
index 8e6e1f3b6..3c8e29981 100644
--- a/src/gpu/jit/gemm/kernel_catalog.hpp
+++ b/src/gpu/jit/gemm/kernel_catalog.hpp
@@ -20,6 +20,7 @@
 #include <string>
 #include <tuple>
 #include <vector>
+#include <cstdint>
 
 #include "gen_gemm_kernel_common.hpp"
 
