From efedbf9080c19241c2aa9ee7ba901245d38c8fa2 Mon Sep 17 00:00:00 2001
From: Naveen Saini <naveen.kumar.saini@intel.com>
Date: Mon, 7 Mar 2022 16:44:24 +0800
Subject: [PATCH 1/2] CMakeLists.txt: exclude host system headers

Upstream-Status: Inappropriate

Signed-off-by: Naveen Saini <naveen.kumar.saini@intel.com>
---
 sources/ippcp/crypto_mb/CMakeLists.txt | 1 -
 1 file changed, 1 deletion(-)

diff --git a/sources/ippcp/crypto_mb/CMakeLists.txt b/sources/ippcp/crypto_mb/CMakeLists.txt
index c4cc82a..4094f34 100644
--- a/sources/ippcp/crypto_mb/CMakeLists.txt
+++ b/sources/ippcp/crypto_mb/CMakeLists.txt
@@ -82,7 +82,6 @@ include_directories(
      ${CRYPTO_MB_INCLUDE_DIR}
      ${OPENSSL_INCLUDE_DIR}
      $<$<C_COMPILER_ID:Intel>:$ENV{ROOT}/compiler/include $ENV{ROOT}/compiler/include/icc>
-     $<$<NOT:$<C_COMPILER_ID:Intel>>:${CMAKE_SYSTEM_INCLUDE_PATH}>
      $<$<OR:$<C_COMPILER_ID:Intel>,$<BOOL:${MSVC_IDE}>>:$ENV{INCLUDE}>
 )
 
-- 
2.17.1

