From e071d6a6323f04dc7e6e40047199e509536d7f6e Mon Sep 17 00:00:00 2001
From: Naveen Saini <naveen.kumar.saini@intel.com>
Date: Mon, 30 Aug 2021 17:54:17 +0800
Subject: [PATCH] demos: use gflags from meta-oe

Upstream-Status: Inappropriate

Signed-off-by: Anuj Mittal <anuj.mittal@intel.com>
Signed-off-by: Naveen Saini <naveen.kumar.saini@intel.com>

---
 demos/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/demos/CMakeLists.txt b/demos/CMakeLists.txt
index 98de0db6c..077d43fb4 100644
--- a/demos/CMakeLists.txt
+++ b/demos/CMakeLists.txt
@@ -140,7 +140,7 @@ endmacro()
 find_package(OpenCV REQUIRED COMPONENTS core highgui videoio imgproc imgcodecs gapi)
 find_package(OpenVINO REQUIRED COMPONENTS Runtime)
 
-add_subdirectory(thirdparty/gflags)
+#add_subdirectory(thirdparty/gflags)
 add_subdirectory(common/cpp)
 add_subdirectory(common/cpp_gapi)
 add_subdirectory(multi_channel_common/cpp)
