From fe3e1d923399da1d71458f5c468563b61784380c Mon Sep 17 00:00:00 2001
From: Naveen Saini <naveen.kumar.saini@intel.com>
Date: Fri, 21 Oct 2022 11:38:23 +0800
Subject: [PATCH] demos: use gflags from meta-oe

Upstream-Status: Inappropriate

Signed-off-by: Anuj Mittal <anuj.mittal@intel.com>
Signed-off-by: Naveen Saini <naveen.kumar.saini@intel.com>
---
 demos/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/demos/CMakeLists.txt b/demos/CMakeLists.txt
index 1484a50d5..996934310 100644
--- a/demos/CMakeLists.txt
+++ b/demos/CMakeLists.txt
@@ -140,7 +140,7 @@ endmacro()
 find_package(OpenCV REQUIRED COMPONENTS core highgui videoio imgproc imgcodecs)
 find_package(OpenVINO REQUIRED COMPONENTS Runtime)
 
-add_subdirectory(thirdparty/gflags)
+#add_subdirectory(thirdparty/gflags)
 add_subdirectory(common/cpp)
 # TODO: remove wrapping if after OpenCV3 is dropped
 if(OpenCV_VERSION VERSION_GREATER_EQUAL 4.5.3)
-- 
2.25.1

