From 1f58c22992ddea4167b01b44448528de427f50d5 Mon Sep 17 00:00:00 2001
From: Dongwon Kim <dongwon.kim@intel.com>
Date: Wed, 2 Mar 2022 15:52:45 -0800
Subject: [PATCH] external ocloc

Upstream-Status: Inappropriate

Signed-off-by: Dongwon Kim <dongwon.kim@intel.com>
---
 cmake/ocloc_cmd_prefix.cmake | 14 ++++++++------
 1 file changed, 8 insertions(+), 6 deletions(-)

diff --git a/cmake/ocloc_cmd_prefix.cmake b/cmake/ocloc_cmd_prefix.cmake
index 2b44330831..03067c9df0 100644
--- a/cmake/ocloc_cmd_prefix.cmake
+++ b/cmake/ocloc_cmd_prefix.cmake
@@ -4,12 +4,14 @@
 # SPDX-License-Identifier: MIT
 #
 
-if(WIN32)
-  set(ocloc_cmd_prefix ocloc)
-else()
-  if(DEFINED NEO__IGC_LIBRARY_PATH)
-    set(ocloc_cmd_prefix ${CMAKE_COMMAND} -E env "LD_LIBRARY_PATH=${LD_LIBRARY_PATH}:${NEO__IGC_LIBRARY_PATH}:$<TARGET_FILE_DIR:ocloc_lib>" $<TARGET_FILE:ocloc>)
+if(NOT DEFINED ocloc_cmd_prefix)
+  if(WIN32)
+    set(ocloc_cmd_prefix ocloc)
   else()
-    set(ocloc_cmd_prefix ${CMAKE_COMMAND} -E env "LD_LIBRARY_PATH=${LD_LIBRARY_PATH}:$<TARGET_FILE_DIR:ocloc_lib>" $<TARGET_FILE:ocloc>)
+    if(DEFINED NEO__IGC_LIBRARY_PATH)
+      set(ocloc_cmd_prefix LD_LIBRARY_PATH=${LD_LIBRARY_PATH}:${NEO__IGC_LIBRARY_PATH}:$<TARGET_FILE_DIR:ocloc_lib> $<TARGET_FILE:ocloc>)
+    else()
+      set(ocloc_cmd_prefix LD_LIBRARY_PATH=${LD_LIBRARY_PATH}:$<TARGET_FILE_DIR:ocloc_lib> $<TARGET_FILE:ocloc>)
+    endif()
   endif()
 endif()
-- 
2.37.3

