From 3c485ba902da0a4a2e9fc43b34eb9174c739929e Mon Sep 17 00:00:00 2001
From: "Yew, Chang Ching" <chang.ching.yew@intel.com>
Date: Fri, 21 May 2021 14:11:44 +0800
Subject: [PATCH] Add X11 support to legacy tools

Upstream-Status: Pending

Signed-off-by: jinchung <jin.chung.teng@intel.com>
Signed-off-by: Yew, Chang Ching <chang.ching.yew@intel.com>
---
 tools/legacy/sample_common/CMakeLists.txt | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/tools/legacy/sample_common/CMakeLists.txt b/tools/legacy/sample_common/CMakeLists.txt
index 5cf69d3..db4de3d 100644
--- a/tools/legacy/sample_common/CMakeLists.txt
+++ b/tools/legacy/sample_common/CMakeLists.txt
@@ -32,6 +32,7 @@ if(UNIX)
     src/vm/time_linux.cpp
     src/vm/atomic_linux.cpp
     src/vm/thread_linux.cpp
+    src/vaapi_utils_x11.cpp
     src/vaapi_utils_drm.cpp)
 else()
   list(
@@ -67,6 +68,8 @@ if(UNIX)
   find_package(PkgConfig REQUIRED)
   pkg_check_modules(PKG_LIBVA libva libva-drm)
   pkg_check_modules(PKG_LIBDRM libdrm)
+  pkg_check_modules(PKG_X11 x11)
+  pkg_check_modules(PKG_LIBVA_X11 libva-x11>=1.10.0)
   if(PKG_LIBVA_FOUND)
     target_compile_definitions(${TARGET} PUBLIC -DLIBVA_SUPPORT)
     target_include_directories(${TARGET} PUBLIC ${PKG_LIBVA_INCLUDE_DIRS})
@@ -79,6 +82,12 @@ if(UNIX)
   else()
     message(WARNING "libva-drm not found: building without libVA DRM support")
   endif()
+  if(PKG_X11_FOUND)
+    target_compile_definitions(${TARGET} PUBLIC -DLIBVA_X11_SUPPORT)
+    target_include_directories(${TARGET} PUBLIC ${PKG_X11_LIBRARY_DIRS})
+  else()
+    message(WARNING "x11 modules not found: building without libVA X11 support")
+  endif()
 
   target_link_libraries(${TARGET} ${CMAKE_DL_LIBS})
 
-- 
2.31.1

