From 79eea3016649aa9e33e564b6971473ac15fa601a Mon Sep 17 00:00:00 2001
From: Naveen Saini <naveen.kumar.saini@intel.com>
Date: Tue, 27 Apr 2021 17:09:01 +0800
Subject: [PATCH] cmake: Allow build env to set variables

Upstream-Status: Inappropriate

Signed-off-by: Naveen Saini <naveen.kumar.saini@intel.com>
---
 cmake/oneAPIInstallDirs.cmake | 32 ++++++++++++++++++++++----------
 1 file changed, 22 insertions(+), 10 deletions(-)

diff --git a/cmake/oneAPIInstallDirs.cmake b/cmake/oneAPIInstallDirs.cmake
index 184e19b..0773500 100644
--- a/cmake/oneAPIInstallDirs.cmake
+++ b/cmake/oneAPIInstallDirs.cmake
@@ -8,23 +8,35 @@
 # Set installation directories
 #
 
-if("${CMAKE_SIZEOF_VOID_P}" STREQUAL "4")
-  set(CMAKE_INSTALL_BINDIR bin32)
-  set(CMAKE_INSTALL_LIBDIR lib32)
-else()
-  set(CMAKE_INSTALL_BINDIR bin)
-  set(CMAKE_INSTALL_LIBDIR lib)
+if(NOT DEFINED CMAKE_INSTALL_LIBDIR)
+    set(CMAKE_INSTALL_LIBDIR lib)
+endif()
+
+if(NOT DEFINED CMAKE_INSTALL_BINDIR)
+   set(CMAKE_INSTALL_BINDIR bin)
+endif()
+
+if(NOT DEFINED CMAKE_INSTALL_ENVDIR)
+    set(CMAKE_INSTALL_ENVDIR env)
+endif()
+
+if(NOT DEFINED CMAKE_INSTALL_MODDIR)
+   set(CMAKE_INSTALL_MODDIR modulefiles)
+endif()
+
+if(NOT CMAKE_INSTALL_EXAMPLEDIR)
+   set(CMAKE_INSTALL_EXAMPLEDIR examples)
+endif()
+
+if(NOT CMAKE_INSTALL_LICENSEDIR)
+    set(CMAKE_INSTALL_LICENSEDIR licensing)
 endif()
 
 set(CMAKE_INSTALL_INCLUDEDIR include)
 set(CMAKE_INSTALL_DOCDIR documentation)
-set(CMAKE_INSTALL_ENVDIR env)
-set(CMAKE_INSTALL_MODDIR modulefiles)
 set(CMAKE_INSTALL_SYSCHECKDIR sys_check)
 set(CMAKE_INSTALL_MODFILEDIR modulefiles)
 set(CMAKE_INSTALL_REDISTDIR redist)
-set(CMAKE_INSTALL_EXAMPLEDIR examples)
-set(CMAKE_INSTALL_LICENSEDIR licensing)
 set(CMAKE_INSTALL_TOOLSDIR tools)
 set(CMAKE_INSTALL_SYSCONFDIR config)
 set(CMAKE_INSTALL_SRCDIR src)
-- 
2.17.1

