From 92d702f885ae7bbfcd5b480c262db805ccc580bf Mon Sep 17 00:00:00 2001
From: John Machado <john.machado@intel.com>
Date: Thu, 16 Nov 2023 12:25:16 +0530
Subject: [PATCH] Add more DG2 Device IDs

Upstream-Status: Backport [https://github.com/intel/gmmlib/commit/92d702f885ae7bbfcd5b480c262db805ccc580bf]

Signed-off-by: Signed-off-by: Lim Siew Hoon <siew.hoon.lim@intel.com>
---
 Source/inc/common/igfxfmid.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/Source/inc/common/igfxfmid.h b/Source/inc/common/igfxfmid.h
index 2b50de2..5ef90f5 100644
--- a/Source/inc/common/igfxfmid.h
+++ b/Source/inc/common/igfxfmid.h
@@ -1928,6 +1928,10 @@ typedef enum __NATIVEGTTYPE
 #define DEV_ID_56B1                             0x56B1
 #define DEV_ID_56B2                             0x56B2
 #define DEV_ID_56B3                             0x56B3
+#define DEV_ID_56BA                             0x56BA
+#define DEV_ID_56BB                             0x56BB
+#define DEV_ID_56BC                             0x56BC
+#define DEV_ID_56BD                             0x56BD
 #define DEV_ID_56C0                             0x56C0
 #define DEV_ID_56C1                             0x56C1
 
-- 
2.40.1

